src/components/InterfaceEditor.tsx

```diff
@@ -270,14 +270,61 @@ const checkInterfaceHasMajorChanges = (
     const draftMapping = draftInterface.mappings.find(
       ({ endpoint }) => endpoint === initialMapping.endpoint,
     );
-    return _.isEqual(draftMapping, initialMapping);
+
+    const sensitiveMappingProperties: Array<keyof AstarteMapping> = [
+      'endpoint',
+      'type',
+      'allowUnset',
+      'reliability',
+      'databaseRetentionPolicy',
+      'databaseRetentionTtl',
+    ];
+    if (!draftMapping) {
+      return false;
+    }
+    const hasExistingMappingsChange = sensitiveMappingProperties.some(
+      (property) => draftMapping[property] !== initialMapping[property],
+    );
+    if (hasExistingMappingsChange) {
+      return false;
+    }
+    return true;
```

Comment:
The code is returning inverted boolean values.

If `draftMapping` doesn't exist, it means that there are major changes, so it should be:
```typescript
if (!draftMapping) {
  return true;
}
```

If `hasExistingMappingsChange` is `true`, there are major changes. So in place of the second conditional, we can just:
```typescript
return hasExistingMappingsChange;
```